### DISCUSSION TITLE
Upgrate to hyper 1.0

### DISCUSSION THREAD

**Discussion by photino** [actix/actix-web#3206] [Q&A] [answered]
How can we upgrade the dependencies to hyper 1.0？Any plan for supporting it?

**Comment by asonix** [comment]
actix-web does not depend on hyper

**Comment by photino** [comment]
Does it depend on `http` and `http-body`？

**Reply by asonix** [reply]
yes for http. no for http-body. 

specifically `actix-http`, `actix-router` and `awc` have dependencies on `http` and can be updated

**Reply by asonix** [reply]
it would also need to be paired with an update to `h2`, which has a dependency on `http` as well

**Reply by photino** [reply]
Thanks! Since `http` 1.0 has released, we can consider upgrading it to the latest version.

**Reply by asonix** [reply, marked as answer]
so the good news is it seems trivial to update these dependencies, the bad news is it likely is a semver-major change, and won't happen until actix-web 5
